libs/common-services/src/main/java/uk/gov/di/ipv/core/library/tracing/TracingHttpClient.java

```diff
@@ -108,6 +136,15 @@ public Optional<Executor> executor() {
         return baseClient.executor();
     }
 
+    // Synchronize to prevent race conditions when multiple threads attempt to recreate client
+    private synchronized void recreateBaseClientIfNecessary() {
+        if (creationTime.plus(MAX_CLIENT_AGE).isBefore(Instant.now())) {
+            LOGGER.info("Recreating base HTTP client due to age");
+            this.baseClient = getInstrumetedHttpClient();
```

Comment:
This line is the actual functional change here - most of the rest of the diff is just moving public methods above private ones.